python-updates - major updates - 2023-02-27

mweinelt opened the pull request:
Another round of major updates, since it has been two months since the last one.

Jobset:
- https://hydra.nixos.org/jobset/nixpkgs/python-updates#tabs-evaluations

Reports:
- https://malob.github.io/nix-review-tools-reports/nixpkgs:python-updates/nixpkgs_python-updates_1791426.html

Dropped updates:
- elasticsearch 8.x, breaks elasticsearch-dsl, which seems abandoned, but is still widely used
- sphinx >6 fails to import extensions, needs investigation
- bokeh, panel still isn't ready, also https://github.com/NixOS/nixpkgs/pull/215406
- catalogue, 2.1.0 was yanked
- tensorboard 2.12, went out of sync with tensorflow

Needs help:
- aws-sam-translator fails its tests with NoRegionError in botocore
  - disabled all tests for now.

TODO:
- [x] Collect open PRs
- [x] Bulk updates
- [x] Early fixes
- [x] Start Evaluation https://hydra.nixos.org/jobset/nixpkgs/python-updates

---

Closes #218443

fabaff committed:
python310Packages.aeppl: add changelog to meta

fabaff committed:
python310Packages.zwave-js-server-python: add changelog to meta

fabaff committed:
python310Packages.pybalboa: add changelog to meta

fabaff committed:
python310Packages.cmd2: add changelog to meta

SuperSandro2000 committed:
pifpaf: init at 3.1.5

SuperSandro2000 committed:
python310Packages.daiquiri: init at 3.2.1

fabaff committed:
python310Packages.gitpython: 3.1.30 -> 3.1.31

Changelog: https://github.com/gitpython-developers/GitPython/blob/3.1.31/doc/source/changes.rst

fabaff committed:
python310Packages.gitpython: add changelog to meta

fabaff committed:
python310Packages.snitun: add changelog to meta

fabaff committed:
python310Packages.snitun: disable on unsupported Python releases

fabaff committed:
python310Packages.redis: update meta

- update disabled

fabaff committed:
python310Packages.pyairnow: add changelog to meta

fabaff committed:
python310Packages.voluptuous-serialize: add changelog to meta

- disable on unsupported Python releases
- add myself as maintainer
- adjust inputs

fabaff committed:
python310Packages.beautifulsoup4: add changelog to meta

SuperSandro2000 committed:
python310Packages.tornado: run tests by switching to github source

fabaff committed:
python310Packages.prometheus-client: add changelog to meta

mweinelt committed:
python3Packages.reqif: 0.0.8 -> 0.0.27

mweinelt committed:
python3Packages.restview: 3.0.0 -> 3.0.1

mweinelt committed:
python3Packages.reolink: 0.63 -> 0.64

mweinelt committed:
python3Packages.requests_ntlm: 1.1.0 -> 1.2.0

mweinelt committed:
python3Packages.ripe-atlas-cousteau: 1.5.1 -> 2.0.0

mweinelt committed:
python3Packages.rich: 13.3.0 -> 13.3.1

mweinelt committed:
python3Packages.rlax: 0.1.4 -> 0.1.5

mweinelt committed:
python3Packages.rpy2: 3.5.8 -> 3.5.9

mweinelt committed:
python3Packages.sagemaker: 2.126.0 -> 2.135.0

mweinelt committed:
python3Packages.riscof: 1.25.2 -> 1.25.3

mweinelt committed:
python3Packages.riscv-config: 3.5.0 -> 3.5.2

mweinelt committed:
python3Packages.schema-salad: 8.4.20230127112827 -> 8.4.20230213094415

mweinelt committed:
python3Packages.scp: 0.14.4 -> 0.14.5

mweinelt committed:
python3Packages.scikit-build-core: 0.1.5 -> 0.2.0

mweinelt committed:
python3Packages.scancode-toolkit: 31.2.3 -> 31.2.4

mweinelt committed:
python3Packages.seaborn: 0.12.1 -> 0.12.2

mweinelt committed:
python3Packages.saneyaml: 0.5.2 -> 0.6.0

mweinelt committed:
python3Packages.scikit-build: 0.16.4 -> 0.16.7

mweinelt committed:
python3Packages.scikit-hep-testdata: 0.4.25 -> 0.4.28

mweinelt committed:
python3Packages.sh: 1.14.3 -> 2.0.2

mweinelt committed:
python3Packages.skrl: 0.8.0 -> 0.10.1

mweinelt committed:
python3Packages.simplejson: 3.18.1 -> 3.18.3

mweinelt committed:
python3Packages.simple-salesforce: 1.12.2 -> 1.12.3

mweinelt committed:
python3Packages.sexpdata: 0.0.3 -> 0.0.4

mweinelt committed:
python3Packages.shapely: 2.0.0 -> 2.0.1

mweinelt committed:
python3Packages.shellingham: 1.5.0 -> 1.5.0.post1

mweinelt committed:
python3Packages.snitun: 0.31.0 -> 0.33.0

mweinelt committed:
python3Packages.sphinx-codeautolink: 0.12.1 -> 0.15.0

mweinelt committed:
python3Packages.softlayer: 6.1.3 -> 6.1.4

mweinelt committed:
python3Packages.sphinx-autodoc-typehints: 1.19.5 -> 1.23.4

mweinelt committed:
python3Packages.soupsieve: 2.3.2.post1 -> 2.4

mweinelt committed:
python3Packages.spyder: 5.4.1 -> 5.4.2

mweinelt committed:
python3Packages.sphinxext-opengraph: 0.7.5 -> 0.8.1

mweinelt committed:
python3Packages.spsdk: 1.8.0 -> 1.9.0

mweinelt committed:
python3Packages.sphinxcontrib-spelling: 7.7.0 -> 8.0.0

mweinelt committed:
python3Packages.sphinxcontrib-confluencebuilder: 1.9.0 -> 2.0.0

mweinelt committed:
python3Packages.sqltrie: 0.0.28 -> 0.1.0

mweinelt committed:
python3Packages.spyder-kernels: 2.4.1 -> 2.4.2

mweinelt committed:
python3Packages.stevedore: 4.1.1 -> 5.0.0

mweinelt committed:
python3Packages.strawberry-graphql: 0.151.2 -> 0.159.0

mweinelt committed:
python3Packages.starlette: 0.23.1 -> 0.25.0

mweinelt committed:
python3Packages.sqlalchemy-utils: 0.39.0 -> 0.40.0

mweinelt committed:
python3Packages.SQLAlchemy: 1.4.46 -> 2.0.4

https://www.sqlalchemy.org/blog/2023/01/26/sqlalchemy-2.0.0-released/

- Migrate to pyproject format with setuptools.
- Prune support for python older than 3.8.
- Run tests in parallel
- Don't run mypy or profiling tests, reducing build time significantly

mweinelt committed:
python3Packages.ssh-mitm: 3.0.1 -> 3.0.2

mweinelt committed:
python3Packages.tempora: 5.1.0 -> 5.2.1

mweinelt committed:
python3Packages.tenacity: 8.1.0 -> 8.2.1

mweinelt committed:
python3Packages.tables: 3.7.0 -> 3.8.0

mweinelt committed:
python3Packages.subarulink: 0.7.1 -> 0.7.5

mweinelt committed:
python3Packages.sunpy: 4.1.0 -> 4.1.3

mweinelt committed:
python3Packages.tabledata: 1.3.0 -> 1.3.1

mweinelt committed:
python3Packages.tensorflow-datasets: 4.8.0 -> 4.8.2

mweinelt committed:
python3Packages.termcolor: 2.1.1 -> 2.2.0

mweinelt committed:
python3Packages.tifffile: 2022.10.10 -> 2023.2.3

mweinelt committed:
python3Packages.testfixtures: 7.0.4 -> 7.1.0

mweinelt committed:
python3Packages.textual: 0.11.1 -> 0.12.1

mweinelt committed:
python3Packages.tiledb: 0.19.0 -> 0.20.0

mweinelt committed:
python3Packages.timetagger: 22.12.1 -> 23.2.1

mweinelt committed:
python3Packages.transformers: 4.25.1 -> 4.26.1

mweinelt committed:
python3Packages.trainer: 0.0.20 -> 0.0.22

mweinelt committed:
python3Packages.trimesh: 3.20.0 -> 3.20.1

mweinelt committed:
python3Packages.transmission-rpc: 3.4.0 -> 4.1.0

mweinelt committed:
python3Packages.traitlets: 5.8.0 -> 5.9.0

mweinelt committed:
python3Packages.trackpy: 0.5.0 -> 0.6.1

mweinelt committed:
python3Packages.tox: 4.1.0 -> 4.4.6

mweinelt committed:
python3Packages.types-pillow: 9.4.0.13 -> 9.4.0.17

mweinelt committed:
python3Packages.types-toml: 0.10.8.2 -> 0.10.8.5

mweinelt committed:
python3Packages.troposphere: 4.2.0 -> 4.3.0

mweinelt committed:
python3Packages.types-redis: 4.4.0.0 -> 4.5.1.4

mweinelt committed:
python3Packages.types-setuptools: 65.6.0.2 -> 67.4.0.3

mweinelt committed:
python3Packages.trytond: 6.6.1 -> 6.6.5

mweinelt committed:
python3Packages.types-psutil: 5.9.5.7 -> 5.9.5.9

mweinelt committed:
python3Packages.typing-extensions: 4.4.0 -> 4.5.0

mweinelt committed:
python3Packages.virtualenv: 20.17.1 -> 20.19.0

mweinelt committed:
python3Packages.vector: 0.11.0 -> 1.0.0

mweinelt committed:
python3Packages.uproot: 5.0.2 -> 5.0.3

mweinelt committed:
python3Packages.uptime-kuma-api: 0.9.0 -> 0.10.0

mweinelt committed:
python3Packages.uamqp: 1.6.3 -> 1.6.4

mweinelt committed:
python3Packages.types-typed-ast: 1.5.8.3 -> 1.5.8.4

mweinelt committed:
python3Packages.unearth: 0.7.0 -> 0.7.2

mweinelt committed:
python3Packages.wcwidth: 0.2.5 -> 0.2.6

mweinelt committed:
python3Packages.werkzeug: 2.2.2 -> 2.2.3

mweinelt committed:
python3Packages.watchdog: 2.2.1 -> 2.3.0

mweinelt committed:
python3Packages.websocket-client: 1.4.2 -> 1.5.1

mweinelt committed:
python3Packages.vulture: 2.6 -> 2.7

mweinelt committed:
python3Packages.whatthepatch: 1.0.3 -> 1.0.4

mweinelt committed:
python3Packages.xhtml2pdf: 0.2.8 -> 0.2.9

mweinelt committed:
python3Packages.xyzservices: 2022.9.0 -> 2023.2.0

mweinelt committed:
python3Packages.xarray: 2022.12.0 -> 2023.2.0

mweinelt committed:
python3Packages.xcffib: 1.1.2 -> 1.2.0

mweinelt committed:
python3Packages.yamllint: 1.28.0 -> 1.29.0

mweinelt committed:
python3Packages.zc.lockfile: 2.0 -> 3.0

mweinelt committed:
python3Packages.zarr: 2.13.3 -> 2.14.2

mweinelt committed:
python3Packages.yark: 1.2.4 -> 1.2.7

mweinelt committed:
python3Packages.yangson: 1.4.15 -> 1.4.16

mweinelt committed:
python3Packages.yq: 3.1.0 -> 3.1.1

mweinelt committed:
python3Packages.yaspin: 2.2.0 -> 2.3.0

mweinelt committed:
python3Packages.zope.proxy: 4.6.1 -> 5.0.0

mweinelt committed:
python3Packages.zstd: 1.5.2.6 -> 1.5.4.0

mweinelt committed:
python3Packages.zstandard: 0.19.0 -> 0.20.0

mweinelt committed:
python3Packages.zope.component: 5.0.1 -> 5.1.0

mweinelt committed:
python3Packages.zope.schema: 6.2.1 -> 7.0.1

mweinelt committed:
python3Packages.zipp: 3.11.0 -> 3.15.0

mweinelt committed:
python310Packages.sphinxeomji: Reformat, use sphinxHook

SuperSandro2000 commented:
> sphinx fails to import certain extension on their latest version @SuperSandro2000
> * sphinx-applehelp 1.0.4 (reverted)
> * sphinx-htmlhelp 2.0.1 (reverted)

That is already the case for certain other extensions. I don't know how to properly fix it but adding them to the sphinx build itself somehow made it work. Probably sphinx does not handle well our usecase.

mweinelt committed:
Revert "sphinx: 5.3.0 -> 6.1.3"

This reverts commit 96fc61bb781dbf604f303088fbeca7fe56512426.

Sphinx versions >=6.0 fail to import extensions, which represents a
severe regression. Details about this issue have been posted at
https://github.com/NixOS/nixpkgs/pull/210338#issuecomment-1447063024.

mweinelt committed:
python310Packages.sphinx: Disable failing tests

mweinelt committed:
python310Packages.sphinx-autoapi: Rename from sphinxcontrib-autoapi

mweinelt committed:
python310Packages.sphinx-autoapi: Disable failing tests

mweinelt committed:
python310Packages.mypy: 0.991 -> 1.0.1

https://mypy-lang.blogspot.com/2023/02/mypy-10-released.html

- Organize dependencies
- Apply patch for setuptools 67.4.0 support
- Use env to pass environment vars explicitly
- Enable tests, disable tests that ignore our PYTHONPATH

fabaff committed:
python310Packages.orjson: 3.8.4 -> 3.8.6

Diff: https://github.com/ijl/orjson/compare/refs/tags/3.8.4...3.8.6

Changelog: https://github.com/ijl/orjson/blob/3.8.6/CHANGELOG.md

fabaff committed:
python310Packages.hahomematic: add orjson

mweinelt committed:
Revert "python3Packages.pyzmq: 24.0.1 -> 25.0.0"

This reverts commit bcc283296af273efec5ad7e4653edea1d7dd7965.

mweinelt committed:
python310Packages.django-extensions: Fix tests with pygments 2.14

SuperSandro2000 committed:
python310Packages.pysaml2: 7.2.1 -> 7.4.1

mweinelt committed:
python310Package.myst-parser: Disable failing tests

fabaff committed:
python310Packages.billiard: add changelog to meta

fabaff committed:
python310Packages.apprise: update license

mweinelt committed:
python310Packages.boto3: 1.26.38 -> 1.26.79

Diff: https://github.com/boto/boto3/compare/1.26.38...1.26.79

Changelog: https://github.com/boto/boto3/blob/1.26.79/CHANGELOG.rst

dotlambda committed:
pytrainer: pin sqlalchemy at 1.4.46

Sqlalchemy-migrate is incompatible with SQLAlchemy 2.0.
Also pin to python310 because sqlalchemy-migrate is incompatible with
Python 3.11.

dotlambda committed:
python310Packages.sqlalchemy-migrate: mark broken

When using SQLAlchemy 2.0 it fails with

    ModuleNotFoundError: No module named 'sqlalchemy.databases'

dotlambda committed:
python310Packages.safety: relax packaging version constraint

kirillrdy committed:
awsebcli: fixup, bump pathspec and termcolor versions

kirillrdy commented:
fix awsebcli https://github.com/NixOS/nixpkgs/pull/218872

Commit:
python3Packages.curio: Fix Linux build

In one of our internal mass rebuilds at work we noticed this package
fail on Linux and the two tests that failed were exactly the ones that
had already been disabled on macOS, so this also disables them on
Linux.

SuperSandro2000 committed:
python310Packages.flask-limiter: normalize pname, complete meta

mweinelt committed:
home-assistant: Relax pip, voluptuous-serialize, override sqlalchemy

mweinelt committed:
grpc: 1.51.1 -> 1.51.2

https://github.com/grpc/grpc/releases/tag/v1.52.1

mweinelt committed:
python310Packages.watchdog: Disable failing test on aarch64

```
__________________________________ test_close __________________________________

    @pytest.mark.skipif(not platform.is_linux(), reason="FileCloseEvent only supported in GNU/Linux")
    @pytest.mark.flaky(max_runs=5, min_passes=1, rerun_filter=rerun_filter)
    def test_close():
        f_d = open(p('a'), 'a')
        start_watching()
        f_d.close()

        # After file creation/open in append mode
        event = event_queue.get(timeout=5)[0]
        assert event.src_path == p('a')
        assert isinstance(event, FileClosedEvent)

        event = event_queue.get(timeout=5)[0]
        assert os.path.normpath(event.src_path) == os.path.normpath(p(''))
        assert isinstance(event, DirModifiedEvent)

        # After read-only, only IN_CLOSE_NOWRITE is emitted but not caught for now #747
        open(p('a'), 'r').close()

>       assert event_queue.empty()
E       assert False
E        +  where False = <bound method Queue.empty of <queue.Queue object at 0xfffff5a96170>>()
E        +    where <bound method Queue.empty of <queue.Queue object at 0xfffff5a96170>> = <queue.Queue object at 0xfffff5a96170>.empty

event      = <DirModifiedEvent: event_type=modified, src_path='/build/tmp99ozlcxp', is_directory=True>
f_d        = <_io.TextIOWrapper name='/build/tmp99ozlcxp/a' mode='a' encoding='UTF-8'>

tests/test_emitter.py:153: AssertionError
```

mweinelt committed:
python310Packages.readme_renderer: Disable failing tests

Likely due to pygments 2.14.0 incompatibiliy. Reported upstream.

mweinelt committed:
python310Packages.kombu: Drop faulty version specifier in requirements

> Executing setuptoolsBuildPhase
> /nix/store/9z2cm958bxarf7n5i48czm7p7lwf9s58-python3.10-setuptools-67.4.0/lib/python3.10/site-packages/setuptools/config/setupcfg.py:520: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
>   warnings.warn(msg, warning_class)
> error in kombu setup command: 'tests_require' must be a string or list of strings containing valid project/version requirement specifiers; Expected end or semicolon (after name and no valid version specifier)
>     pytz>dev
>         ^

mweinelt committed:
python310Packages.numba: Backport support for numpy 1.24.x

This is a last ditch effort to keep numba working.

mweinelt committed:
python310Packages.numcodecs: Disable failing tests

These started failing with numpy 1.24.

mweinelt committed:
python310Packages.send2trash: Fix setup.cfg validation confusion

dotlambda commented:
Looks like we won't be able to make Numba work with NumPy 1.24: https://github.com/numba/numba/issues/8464
cc @FRidh

FRidh commented:
Then we should wait again with the numpy update.

fabaff committed:
python310Packages.requests_ntlm: add changelog to meta

fabaff committed:
python310Packages.send2trash: add changelog to meta

fabaff committed:
python310Packages.moto: add changelog to meta

fabaff committed:
python310Packages.tabledata: add changelog to meta

fabaff committed:
python310Packages.yamllint: add changelog to meta

fabaff committed:
python310Packages.yamllint: update disabled

fabaff committed:
python310Packages.requests_ntlm: normalize pname

fabaff committed:
python310Packages.tifffile: add changelog to meta

fabaff committed:
python310Packages.tifffile: update disabled

fabaff committed:
python310Packages.zarr: add changelog to meta

fabaff committed:
python310Packages.zarr: update disabled and add format

fabaff committed:
python310Packages.zarr: switch to pytestCheckHook

- add pythonImportsCheck

fabaff committed:
python310Packages.pythonfinder: add changelog to meta

fabaff committed:
python310Packages.pythonfinder: add pythonImportsCheck

fabaff committed:
python310Packages.pytest-localserver: add changelog to meta

fabaff committed:
python310Packages.rich: add changelog to meta

fabaff committed:
python310Packages.pytest-httpx: add changelog to meta

SuperSandro2000 commented on pkgs/development/python-modules/pydocstyle/default.nix:
```diff
@@ -1,32 +1,44 @@
+
   propagatedBuildInputs = [
     snowballstemmer
+  ] ++ lib.optionals (pythonOlder "3.11") [
+    tomli
+  ];
+
+  passthru.optional-dependencies.toml = [
+    tomli
```
This is duplicated, we could just condense this to always include tomli

SuperSandro2000 committed:
python310Packages.babel: 2.11.0 -> 2.12.1

mweinelt committed:
python310Packages.databases: Mark broken with sqlalchemy>=2.0

The upstream indicated that they have no resources to develop support
themselves.

mweinelt committed:
hatch: Disable failing test

mweinelt committed:
python310Packages.gruut: Relax networkx constraints

mweinelt committed:
python310Packages.aiopyarr: Substitute version value

Because setuptools has a stricter validation now, this started
breaking.

> setuptools.extern.packaging.version.InvalidVersion: Invalid version: 'master'

mweinelt committed:
python310Packages.pylint: 2.15.9 -> 2.16.2

https://github.com/PyCQA/pylint/releases/tag/v2.15.10
https://github.com/PyCQA/pylint/releases/tag/v2.16.0
https://github.com/PyCQA/pylint/releases/tag/v2.16.1
https://github.com/PyCQA/pylint/releases/tag/v2.16.2

The manpage now needs to be built with sphinx, but we lack certain
extensions like sphinx-reredirect.

The elisp minor mode files now live at
https://github.com/emacsorphanage/pylint.

mweinelt committed:
pinnwand: Relax sqlalchemy constraint

Package test and integration tests are passing and I reported that
success story upstream.

mweinelt committed:
python310Packages.flask-login: Fix tests by passing asgiref

mweinelt committed:
python310Packages.prompt-toolkit: 3.0.37 -> 3.0.38

https://github.com/prompt-toolkit/python-prompt-toolkit/blob/3.0.38/CHANGELOG

dotlambda committed:
python310Packages.dropbox: fix build

dotlambda committed:
python310Packages.markdown2: Use hash, add changelog

SuperSandro2000 committed:
rav1e: fix build

zlib.h header was missing and because of that compilation failed.

SuperSandro2000 committed:
python310Packages.flask-babel: fix build against babel 12.2.X

see https://github.com/python-babel/flask-babel/issues/221
and https://github.com/python-babel/flask-babel/pull/222

dotlambda commented on pkgs/development/python-modules/flask-babel/default.nix:
```diff
@@ -32,6 +32,10 @@ buildPythonPackage rec {
   };
 
+  patches = [
+    ./fix-tests-with-babel-12-2.diff
```
@SuperSandro2000 You forgot to commit the file.

mweinelt committed:
python310Packages.django-celery-results: Fix build

The `tests_require` value violates setuptools' expectations.

error in django_celery_results setup command: 'tests_require' must be a
string or list of strings containing valid project/version requirement
specifiers; Expected end or semicolon (after name and no valid version
specifier)

mweinelt commented on pkgs/top-level/python-aliases.nix:
```diff
@@ -50,6 +50,7 @@ mapAliases ({
   bt_proximity = bt-proximity; # added 2021-07-02
   BTrees = btrees; # added 2023-02-19
   carrot = throw "carrot has been removed, as its development was discontinued in 2012"; # added 2022-01-18
+  cchardet = faust-cchardet; # added 2023-02-14
```
On second thought, won't that bite us with regards to packages relying on aliases?

```
❯ rg cchardet -l
pkgs/development/tools/poetry2nix/poetry2nix/overrides/build-systems.json
pkgs/development/python-modules/cchardet/default.nix
pkgs/development/python-modules/aiohttp/default.nix
pkgs/development/python-modules/surepy/default.nix
pkgs/development/python-modules/nextcord/default.nix
pkgs/development/python-modules/vulcan-api/default.nix
pkgs/development/python-modules/clevercsv/default.nix
pkgs/applications/misc/calibre/default.nix
pkgs/servers/home-assistant/appdaemon.nix
pkgs/tools/backup/grab-site/default.nix
pkgs/tools/security/xcat/default.nix
pkgs/tools/security/cve-bin-tool/default.nix
```

dotlambda replied:
Yes, I forgot to change the references.

fabaff committed:
python310Packages.locationsharinglib: 4.1.8 -> 4.2.0

Changelog: https://github.com/costastf/locationsharinglib/blob/4.2.0/HISTORY.rst

fabaff committed:
python310Packages.pip-requirements-parser: 32.0.0 -> 32.0.1

Diff: https://github.com/nexB/pip-requirements-parser/compare/refs/tags/v32.0.0...v32.0.1

Changelog: https://github.com/nexB/pip-requirements-parser/blob/v32.0.1/CHANGELOG.rst

fabaff committed:
python310Packages.container-inspector: add changelog to meta

fabaff committed:
python310Packages.container-inspector: fix invalid version specifier

fabaff committed:
python310Packages.typecode: add changelog to meta

fabaff committed:
python310Packages.typecode: fix invalid version specifier

fabaff committed:
python310Packages.extractcode: add changelog to meta

fabaff committed:
python310Packages.extractcode: fix invalid version specifier

mweinelt committed:
python310Packages.cchardet: Drop in favor of faust-cchardet

The former packages has seen its last release in 2020-10 and can be
considered abandoned. Meanwhile a new fork has appeared in
faust-cchardet, that we're going to use in its place.

Co-Authored-By: Robert Schütz <nix@dotlambda.de>

mweinelt committed:
tts: update overrides, passthru python

dotlambda committed:
python310Packages.clevercsv: remove superfluous postPatch

mweinelt committed:
python3.pkgs: Migrate fetchers to use hash

when they already rely on SRI hashes.

mweinelt committed:
aws-cli: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
caffeine-ng: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

Also set an explicit format for buildPythonApplication.

mweinelt committed:
catt: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

Also set an explicit format in buildPythonApplication.

mweinelt committed:
changedetection-io: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
checkov: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
csvs-to-sqlite: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

Also set explicit format in buildPythonApplication.

mweinelt committed:
elasticsearch-curator: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
pgadmin: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
python310Packages.decorator: Use SRI hash

mweinelt committed:
flare-floss: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
privacyidea: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
python310Packages.pyuavcan: Use sri hashes and hash

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

mweinelt committed:
docker-compose_1: Use SRI hash

To allow awsebcli to use its changeversion function to uniformly
overwrite the hash.

mweinelt committed:
awsebcli: Use hash and SRI hashes

With the python package set moving to hash, overriding the fetcher and
reintroducing sha256 breaks eval, due to multiple hashes passed.

dotlambda committed:
python310Packages.ipywidgets: skip failing test

mweinelt committed:
python310Packages.decorator: Enable tests

dotlambda committed:
libgit2_1_5: init at 1.5.1

dotlambda committed:
python310Packages.pygit2: use libgit2_1_5

mweinelt committed:
python310Packages.dask: Disable failing test

dotlambda committed:
python310Packages.astroid: add meta.changelog

fabaff committed:
python310Packages.timetagger: update meta

fabaff committed:
python310Packages.timetagger: disable on unsupported Python releases

fabaff committed:
python310Packages.timetagger: add missing inputs

fabaff committed:
python310Pacakges.natsort: add changelog to meta

fabaff committed:
python310Packages.natsort: 8.2.0 -> 8.3.1

Changelog: https://github.com/SethMMorton/natsort/blob/8.3.1/CHANGELOG.md

fabaff committed:
python310Packages.fastnumbers: add changelog to meta

fabaff committed:
python310Packages.fastnumbers: 3.2.1 -> 5.0.1

Changelog: https://github.com/SethMMorton/fastnumbers/blob/5.0.1/CHANGELOG.md

mweinelt committed:
python310Packages.notebook: Disable failing test

mweinelt committed:
python310Packages.nikola: fix src pname, disable failing tests

mweinelt committed:
python310Packages.soxr: Fix aarch64 build

There are patches that gate certain code parts for aarch64 platforms,
that are not part of the soxr release.

mweinelt committed:
python310Packages.zigpy: Disable failing test

mweinelt committed:
python310Packages.flask-babelex: Disable failing tests

set explicit format. This package is unmaintained upstream and should be
removed, but flask-admin and pgadmin depend on it.

mweinelt committed:
python310Packages.jupyter-server-mathjax: Fix build

mweinelt committed:
python310Packages.nbdime: Mark broken with jupyter-server>2

mweinelt committed:
python310Packages.nbval: Fix build

Exclude nbdime tests, since it was marked broken due to jupyter-server 2
incompatibilities.

Also set explicit format.

mweinelt committed:
python310Packages.multiset: Fix build

The package has a broken python_requires version specifier, that breaks
the build with newer setuptools versions.

Convert the package to pep517 build, migrate the fetcher to an SRI hash,
and explicitly enable tests.

mweinelt committed:
python310Packages.seaborn: Disable failing test

and reenable tests that were disabeld for mpl 3.5. Also use
pytest-xdist, as upstream does, because otherwise these tests are taking
ages.

mweinelt committed:
gdal: Disable failing test

```
>           raise CalledProcessError(retcode, cmd)
E           subprocess.CalledProcessError: Command '['/nix/store/r0xi4wqmplkiapxhi8ns7hd3jl5011gc-python3-3.10.10/bin/python3.10', 'osr_basic_subprocess.py']' returned non-zero exit status 1.

/nix/store/r0xi4wqmplkiapxhi8ns7hd3jl5011gc-python3-3.10.10/lib/python3.10/subprocess.py:369: CalledProcessError
----------------------------- Captured stderr call -----------------------------
ERROR 1: PROJ: proj_create_from_database: crs not found
Traceback (most recent call last):
  File "/build/source/autotest/osr/osr_basic_subprocess.py", line 42, in <module>
    assert sr.ImportFromEPSG(4326) == 0
AssertionError
```

mweinelt committed:
python310Packages: Fix build, extract optionals

mweinelt committed:
nixos/tests/mjolnir: Require matrix-nio[e2e]

mweinelt committed:
matrix-commander: Require matrix-nio[e2e]

mweinelt committed:
mirage: Require matrix-nio[e2e]

mweinelt committed:
pantalaimon: Require matrix-nio[e2e]

mweinelt committed:
weechatScripts.weechat-matrix: Require matrix-nio[e2e]

mweinelt committed:
opsdroid: Require matrix-nio[e2e]

mweinelt committed:
python310Packages.delorean: Fix build

Propagate missing pytz dependency.

Also set explicit format, use hash for src fetcher.

mweinelt committed:
python310Packages.btchip-python: Rename from btchip

mweinelt committed:
python310Packages.btchip-python: Fix extra_requires version specifier

And reformat, expose optional-dependencies, set explicit format.

mweinelt committed:
python310Packaegs.mautrix: Disable failing tests

They are incompatible with sqlalchemy>=2

mweinelt committed:
python310Packages.glom: Relax face pin

mweinelt committed:
python310Packages.dj-rest-auth: Disable failing test

Run test phase explicitly, pass django as build input, expose
optional-dependencies.

mweinelt committed:
python310Packages.home-assistant-chip-core: Fix aarch64-linux hash

mweinelt committed:
python310Packages.amaranth: Fix extra_requires version specifier

mweinelt committed:
python310Packages.dask: 2023.1.0 -> 2023.2.1

Diff: https://github.com/dask/dask/compare/2023.1.0...2023.2.1

Changelog: https://docs.dask.org/en/latest/changelog.html

mweinelt committed:
python310Packages.cherrypy: Expose ssl extra

mweinelt committed:
poetry: 1.3.2 -> 1.4.0

https://github.com/python-poetry/poetry/releases/tag/1.4.0

mweinelt committed:
python310Packages.pyhanko-certvalidator: 0.19.5 -> 0.20.1

mweinelt committed:
python310Packages.certomancer: Disable failing test

after pyhanko-certvalidator update.

mweinelt committed:
python310Packages.pyhanko: 0.12.1 -> 0.17.0

https://github.com/MatthiasValvekens/pyHanko/blob/0.17.0/docs/changelog.rst#0170

Drop the custom flags in favor of the standardised
optional-dependencies. Always test with all optional-dependencies.

mweinelt committed:
python310Packages.flask-httpauth: Fix tests

by passing in flask[async].

Also convert to pep517 build, reformat.

danieldk commented on pkgs/development/python-modules/blis/default.nix:
```diff
@@ -12,14 +12,14 @@ buildPythonPackage rec {
   # Do not update to BLIS 0.9.x until the following issue is resolved:
   # https://github.com/explosion/thinc/issues/771#issuecomment-1255825935
-  version = "0.7.9";
+  version = "0.9.1";
```
BLIS 0.9.1 is broken, see: https://github.com/NixOS/nixpkgs/pull/207070

laikq commented:
Please revert or amend commit a8d46a2: package `pypdf2` should not be updated to version 3, it breaks `pdfposter` and probably other things.

mweinelt commented:
As `pdfposter` is not the only consumer of `pypdf2` this needs more investigation to understand if that is feasible. As it stands it cannot be reverted easily.
